Add several LS completion/rename support improvements to release note (#104)
ballerina-platform/ballerina-release: release-notes/swan-lake-alpha3-release-note.md

#### Developer Tools

##### Language Server
###### Completions

> - Completion support for field accesses, varible declarations and assignment expressions are now context aware. i.e: 

Reviewer: ```suggestion
- Completion support for field access, variable declarations, and assignment expressions are now context aware. i.e: 
```

> - Completion support for field access, variable declarations, and assignment expressions are now context aware. i.e: 

Reviewer 2: ```suggestion
- Completion support for field access, variable declarations, and assignment expressions are now context-aware. That means the following.
```

>   - When we declara a vairiable or perform an assignment, suggestions show the methods/variables which return/are values of same type as the variable.

Reviewer: ```suggestion
  - When we declare a variable or perform an assignment, suggestions show the variables which are of same type as the variable and methods that return a value of the same type.
```

>   - When we declare a variable or perform an assignment, suggestions show the variables which are of same type as the variable and methods that return a value of the same type.

Reviewer: ```suggestion
  - When we declare a variable or perform an assignment, suggestions show the variables which are of same type as the variable and functions that return a value of the same type.
```

>   - When we declare a variable or perform an assignment, suggestions show the variables which are of same type as the variable and methods that return a value of the same type.

Reviewer 2: ```suggestion
  - When we declare a variable or perform an assignment, suggestions show the variables, which are of the same type as the variable, and methods that return a value of the same type.
```

>   - Object and record field accesses now suggest object/record fields sorted at the top.

Reviewer: How about
```suggestion
  - Object and record field access now suggests object and record fields first.
```

>   - Object and record field access now suggests object and record fields first.

Reviewer 2: ```suggestion
  - Object and record field access now suggests the object and record fields first.
```

###### Rename

> - Rename now validates the `newText` for a valid identifier before performing the rename

Reviewer: ```suggestion
- Rename now validates if the new name is a valid identifier before performing the rename.
```
